javascript/ql/src/Declarations/SuspiciousMethodNameDeclaration.ql

```diff
@@ -26,45 +26,45 @@ predicate isSuspisousMethodName(string name, ClassOrInterface container) {
   name = "new" and container instanceof ClassDefinition
 }
 
-from MethodDeclaration member, ClassOrInterface container, string suffixMsg
+from MethodDeclaration member, ClassOrInterface container, string name, string msg
 where
   container.getLocation().getFile().getFileType().isTypeScript() and
-  container.getAMember() = member and
-  isSuspisousMethodName(member.getName(), container) and
+  container.getMember(name) = member and
+  isSuspisousMethodName(name, container) and
   
-  // Assume that a "new" method is intentional if the class has an explicit constructor.
+  // Cases to ignore.
   not (
-    member.getName() = "new" and
-    container instanceof ClassDefinition and
-    exists(ConstructorDeclaration constructor |
-      container.getAMember() = constructor and
-      not constructor.isSynthetic() 
+	( // Assume that a "new" method is intentional if the class has an explicit constructor.
+	  name = "new" and
+	  container instanceof ClassDefinition and
+	  not container.getMember("constructor").(ConstructorDeclaration).isSynthetic()
```

Comment:
(ditto the previous comment about a reformulation that did not preserve semantics)